product-service/src/functions/getProductsById/handler.ts

```diff
@@ -0,0 +1,24 @@
+import { APIGatewayProxyResult } from 'aws-lambda/trigger/api-gateway-proxy';
+import { middyfy } from '@libs/lambda';
+import products from '../../products.json';
+import type { ValidatedEventAPIGatewayProxyEvent } from '@libs/api-gateway';
+import { notFoundResponse } from '../../responses';
+
+const getProductsById: ValidatedEventAPIGatewayProxyEvent<APIGatewayProxyResult> = async (event) => {
+  const productId = event?.pathParameters?.id;
+  const product = products.find(product => product.id === productId);
+
+  if(!product){
+    return notFoundResponse('Product not found');
+  }
+  
+  return {
+    headers: { 
+      'Access-Control-Allow-Origin': '*' 
```

Comment:
nit: no need to add this header if you use @middy/http-cors. Valid for all lambda response (200,  404, ...).